internal/pkg/agent/application/coordinator/diagnostics_test.go

```diff
@@ -106,15 +106,14 @@ agent:
         - host1
         - host2
       environment: diag-unit-test
-      apikey: apikey
-      secrettoken: secret
-      globallabels:
+      api_key: apikey
```

Comment:
Oh I think I see now, the `yaml` tag additions in `core/monitoring/config/config.go` are to fix the way the fields are marshaled into Yaml here.

I don't know the context of this config, I wonder if this might be treated as a breaking change? Is it possible that a user currently depends on the `apikey` not being `api_key`, or on `skipverify` to always exist? Or is the structure of this config an internal detail that user's aren't supposed to rely on?

Reply:
Exactly, I already spotted that the APM config fields used different keys when dumped into yaml, and I tried to correct (as much as possible) in this PR

It's not a breaking change as diagnostics are used internally to debug SDHs and weird agent behaviors, the consumer of these files is a human.

I tried to align the keys to facilitate copy/paste from diagnostics to config files/policies when that becomes necessary.

It would be a breaking change if I changed the `config:` tags as that would change the expected keys in config file/policy ;), changing the diagnostics output we are just trying to align those without any breakage :wink:

Reply:
Awesome, thanks for explaining this! :+1: